x/emissions/keeper/keeper.go

```diff
@@ -1108,7 +1112,15 @@ func (k *Keeper) AddDelegateStake(ctx context.Context, topicId TopicId, delegato
 	if err != nil {
 		return err
 	}
-	stakePlacementNew := delegateStakePlacement.Add(stake)
+	newAmount := delegateStakePlacement.Amount.Add(stake)
+	share, err := k.GetDelegateRewardPerShare(ctx, topicId, reputer)
+	if err != nil {
+		return err
+	}
+	stakePlacementNew := types.DelegatorInfo{
+		Amount:     newAmount,
+		RewardDebt: newAmount.Mul(share),
```

Comment:
isn't this combining two different pieces of data? so say share is 1 and later 2, and someone adds 10 at each time. 

first update is amount 10 and reward debt 10, the second time it should be 20 and reward debt 30, but it would update as 20 and reward debt 40, does that make sense?

Reply:
like it should be rewardDebt: rewardDebt + stake * share

Reply:
we calculate the reward like this:
reward = amount * share - rewardDebt

Reply:
I guess asked another way why bother storing the rewardDebt if it's always just the amount multiplied by share? Why not store share instead?

Reply:
And my hunch is the answer is because rewardDebt should be an accumulator that maintains the accumulation of previous amount * share calculations instead of just the current amount * share calculation